Rakefile

```diff
@@ -14,66 +14,50 @@ task :generate_csv do
     "ocd-division/country:ug/" + parts.map { |type, value| [type, idify(value)].join(':') }.join('/')
   end
 
-  csv_url = 'https://docs.google.com/spreadsheets/d/1LetNFNq6ovg4bbq-Whze0Q06CYSpNqpRIzSUb9yVtw4/export?format=csv'
-
-  csv = CSV.parse(open(csv_url).read, headers: true, header_converters: :symbol)
-
-  ocd_ids = Set.new
-
-  csv.each do |r|
-    next unless r[:arearegion]
-    ocd_ids << {
-      id: id_for(region: r[:arearegion]),
-      name: r[:arearegion]
-    }
-
-    next unless r[:areasub_region]
-    ocd_ids << {
-      id: id_for(region: r[:arearegion], subregion: r[:areasub_region]),
-      name: r[:areasub_region]
-    }
-
-    next unless r[:areadistrict]
-    ocd_ids << {
-      id: id_for(region: r[:arearegion], subregion: r[:areasub_region], district: r[:areadistrict]),
-      name: r[:areadistrict]
-    }
-
-    next unless r[:areaconstituency]
-    ocd_ids << {
-      id: id_for(region:r[:arearegion], subregion: r[:areasub_region], district: r[:areadistrict], constituency: r[:areaconstituency]),
-      name: r[:areaconstituency]
-    }
+  def ocd_ids_from_csv(csv_data, mapping = {})
+    csv = CSV.parse(csv_data, headers: true, header_converters: :symbol)
+    ocd_ids = Set.new
+
+    csv.each do |r|
+      region = r[mapping.fetch(:region, :region)]
+      next unless region
+      ocd_ids << {
+        id: id_for(region: region),
+        name: region.strip
+      }
+
+      subregion = r[mapping.fetch(:subregion, :subregion)]
+      next unless subregion
+      ocd_ids << {
+        id: id_for(region: region, subregion: subregion),
+        name: subregion.strip
+      }
+
+      district = r[mapping.fetch(:district, :district)]
+      next unless district
+      ocd_ids << {
+        id: id_for(region: region, subregion: subregion, district: district),
+        name: district.strip
+      }
+
+      constituency = r[mapping.fetch(:constituency, :constituency)]
+      next unless constituency
+      next if ['-', '_'].include?(constituency)
```

Comment:
Ah, this is because [a couple of rows in the corrections CSV](https://github.com/theyworkforyou/uganda_ocd_ids/blob/1255164b9eacd2ad0a00fc4f91914c2e5714dd3e/area_info.csv#L2) have `-` or `_` in them, so this was just to skip them. I guess the simplest thing to do is to just remove those cells from the CSV. Alternatively we could have an extra argument to say what cell values should be skipped.